### PR TITLE
Switch to spdlog

### PR DESCRIPTION
**Pull request by jhale**
There is a lot of tedious work to do, seems like a good point to get feedback.

### CODE REVIEW

File: cpp/dolfin/common/MPI.cpp [hunk 1]
```diff
@@ -17,7 +20,8 @@ dolfin::MPI::Comm::Comm(MPI_Comm comm)
   {
     int err = MPI_Comm_dup(comm, &_comm);
     if (err != MPI_SUCCESS)
-      log::error("Duplication of MPI communicator failed (MPI_Comm_dup");
+      spdlog::error("Duplication of MPI communicator failed (MPI_Comm_dup)");
+      throw std::runtime_error("Duplication of MPI communicator failed (MPI_Comm_dup)");
```
Comment (jhale) [inline cpp/dolfin/common/MPI.cpp]: The old design logged and threw. The new design requires separate steps. Thoughts?

File: cpp/dolfin/mesh/IntervalCell.cpp [hunk 2]
```diff
@@ -26,9 +27,10 @@ std::size_t IntervalCell::num_entities(std::size_t dim) const
   case 1:
     return 1; // cells
   default:
-    log::dolfin_error("IntervalCell.cpp",
-                      "access number of entities of interval cell",
-                      "Illegal topological dimension (%d)", dim);
+    spdlog::error("IntervalCell.cpp: "
+                  "access number of entities of interval cell. "
+                  "Illegal topological dimension ({}).", dim);
+    throw std::invalid_argument();
```
Comment (jhale) [inline cpp/dolfin/mesh/IntervalCell.cpp]: Again, duplicate error text?

File: cpp/dolfin/common/MPI.cpp (hunk 1 above)
Comment (garth-wells) [reply cpp/dolfin/common/MPI.cpp]: I think it looks ok - bare in mind that the logger will (or should) be used for much more than just error logging.

Do we have any indication of what is 'good' practice?

**Comment by chrisrichardson** [comment]
OK, I went through and changed everything to use spdlog. Seems OK, but I'm sure there are some rough edges to be tidied up.

**Pull request merged**